Update react peer dependency to >=19.0.0

aminabromand opened the pull request:
`npm run test` was successful

ai commented on package.json:
```diff
@@ -37,7 +37,7 @@
   "peerDependencies": {
     "nanostores": "^0.9.0 || ^0.10.0 || ^0.11.0",
-    "react": ">=18.0.0"
+    "react": ">=19.0.0"
```
By this change you will break all React 18 users

aminabromand commented:
> `>=18.0.0` already contain `19.0.0`. Note that we use `>=` and not `^`

when I create a new `next` project and try to install `@nanostores/react` I get this output:
```
amin@tux ~/dev/test/blub $ npx create-next-app@latest
✔ What is your project named? … my-app
✔ Would you like to use TypeScript? … No / Yes
✔ Would you like to use ESLint? … No / Yes
✔ Would you like to use Tailwind CSS? … No / Yes
✔ Would you like your code inside a `src/` directory? … No / Yes
✔ Would you like to use App Router? (recommended) … No / Yes
✔ Would you like to use Turbopack for next dev? … No / Yes
✔ Would you like to customize the import alias (@/* by default)? … No / Yes
Creating a new Next.js app in /home/amin/dev/test/blub/my-app.

Using npm.

Initializing project with template: app-tw


Installing dependencies:
- react
- react-dom
- next

Installing devDependencies:
- typescript
- @types/node
- @types/react
- @types/react-dom
- postcss
- tailwindcss
- eslint
- eslint-config-next

npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.

added 372 packages, and audited 373 packages in 12s

138 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
Initialized a git repository.

Success! Created my-app at /home/amin/dev/test/blub/my-app

amin@tux ~/dev/test/blub $ cd my-app/
amin@tux ~/dev/test/blub/my-app $ npm install nanostores

added 4 packages, and audited 377 packages in 1s

141 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
amin@tux ~/dev/test/blub/my-app $ npm install @nanostores/react
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: my-app@0.1.0
npm error Found: react@19.0.0-rc-66855b96-20241106
npm error node_modules/react
npm error   react@"19.0.0-rc-66855b96-20241106" from the root project
npm error
npm error Could not resolve dependency:
npm error peer react@">=18.0.0" from @nanostores/react@0.8.0
npm error node_modules/@nanostores/react
npm error   @nanostores/react@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /home/amin/.npm/_logs/2024-11-27T18_29_45_735Z-eresolve-report.txt
npm error A complete log of this run can be found in: /home/amin/.npm/_logs/2024-11-27T18_29_45_735Z-debug-0.log
amin@tux ~/dev/test/blub/my-app $
```

ai commented:
Maybe it is because of `-rc`? It is a special version which could not match `>=18.0.0`.

Anyway, breaking all React 18 users is more damage,

aminabromand commented:
You are right, it must not break React 18 users. And also the `rc` is causing the trouble.

So I have created a new pull request. As far as I understand, the new request should not break React 18 users, but allow for React 19 rc versions.
